pkg/validators/am0010_k8s_resource_names.go

```diff
@@ -0,0 +1,234 @@
+package validators
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/mt-sre/addon-metadata-operator/pkg/types"
+	"k8s.io/apimachinery/pkg/api/validation"
+	utilvalidation "k8s.io/apimachinery/pkg/util/validation"
+)
+
+var AM0010 = types.Validator{
+	Code:        "AM0010",
+	Name:        "k8s_resource_and_field_names",
+	Description: "Validates k8s namespaces, labels, and annotations within Addon metadata against k8s standards",
+	Runner:      Validatek8sResourceAndFieldNames,
+}
+
+func init() {
+	Registry.Add(AM0010)
+}
+
+const (
+	emptyMsg = failureMsg("")
+)
+
+type failureMsg string
+
+func (m failureMsg) IsEmpty() bool {
+	return m == ""
+}
+
+func prefixedFailureMsg(prefix string, msg failureMsg) failureMsg {
+	return failureMsg(fmt.Sprintf("%s: %s", prefix, msg))
+}
+
+func joinFailureMsgs(msgs ...failureMsg) failureMsg {
+	msgStrings := make([]string, 0, len(msgs))
+
+	for _, m := range msgs {
+		msgStrings = append(msgStrings, string(m))
+	}
+
+	return failureMsg(strings.Join(msgStrings, ", "))
+}
```

Comment:
This could be moved to `pkg/validators/result.go` or `pkg/validators/helpers.go`

Reply:
Yes, I wanted to leave this here for now because I'd like to look at more extensive changes to `ValidatorResult` in a future PR. Basically I hate the check-fix loop that happens when checkers don't report every failure the first time which is why I have this aggregation logic to join failures msgs. However, I'd like to update `ValidatorResult.FailureMsg` from a `string` to a `[]string` and then apply some of the helpers behind the scenes. A little too much for this PR though IMO.